shared/chat/conversation/list-area/normal/index.desktop.js

```diff
@@ -173,27 +219,60 @@ class Thread extends React.PureComponent<Props, State> {
       return
     }
 
-    // we send something last
+    // someone requested we scroll to bottom and lock (ignore if we don't have latest)
     if (
-      this.props.messageOrdinals.last() !== prevProps.messageOrdinals.last() &&
-      this.props.lastMessageIsOurs
+      this.props.scrollListToBottomCounter !== prevProps.scrollListToBottomCounter &&
+      this.props.containsLatestMessage
     ) {
-      this.setState(p => (p.isLockedToBottom ? null : {isLockedToBottom: true}))
-      this._scrollToBottom('componentDidUpdate-sent-something')
+      this.setState(p => (p.lockedToBottom ? null : {lockedToBottom: true}))
+      this._scrollToBottom('componentDidUpdate-requested')
       return
     }
 
-    // Adjust scrolling
+    // Adjust scrolling if locked to the bottom
     const list = this._listRef.current
-    // Prepending some messages?
-    if (snapshot && list && !this.state.isLockedToBottom) {
+    // if locked to the bottom, and we have the most recent message, then scroll to the bottom
+    if (this._isLockedToBottom() && this.props.conversationIDKey === prevProps.conversationIDKey) {
+      // maintain scroll to bottom?
+      this._scrollToBottom('componentDidUpdate-maintain-scroll')
+    }
+
+    // Check if we just added new messages from the future. In this case, we don't want to adjust scroll
+    // position at all, so just bail out if we detect this case.
+    if (
+      this.props.messageOrdinals.size !== prevProps.messageOrdinals.size &&
+      this.props.messageOrdinals.first() === prevProps.messageOrdinals.first() &&
+      !this.props.containsLatestMessage
+    ) {
+      // do nothing do scroll position if this is true
+      this._scrollHeight = 0 // setting this causes us to skip next resize
+      return
+    }
+
+    // Check to see if we jumped to some other set of ordinals, or that our centered ordinal is different
+    // than what we scroll to previously
+    if (
+      !!this.props.centeredOrdinal &&
+      (this.props.centeredOrdinal !== this._lastCenteredOrdinal ||
```

Comment:
any reason not to check against `prevProps.centeredOrdinal`? then you don't need `this._lastCenteredOrdinal`

Reply:
I ran into a problem where the centeredOrdinal would change, but the ordinals list was the same and nothing would happen. In other words, there were two updates, one for the centered ordinal, and another for the new list of messages. I wanted both to be in place before doing the scroll.